cluster-autoscaler/utils/drain/drain.go

```diff
@@ -106,104 +106,118 @@ func GetPodsForDeletionOnNodeDrain(
 			continue
 		}
 
-		isDaemonSetPod := false
 		replicated := false
 		safeToEvict := hasSafeToEvictAnnotation(pod)
 		terminal := isPodTerminal(pod)
 
 		controllerRef := ControllerRef(pod)
 		refKind := ""
-		if controllerRef != nil {
-			refKind = controllerRef.Kind
-		}
 
-		// For now, owner controller must be in the same namespace as the pod
-		// so OwnerReference doesn't have its own Namespace field
-		controllerNamespace := pod.Namespace
-
-		if refKind == "ReplicationController" {
-			if checkReferences {
-				rc, err := listers.ReplicationControllerLister().ReplicationControllers(controllerNamespace).Get(controllerRef.Name)
-				// Assume a reason for an error is because the RC is either
-				// gone/missing or that the rc has too few replicas configured.
-				// TODO: replace the minReplica check with pod disruption budget.
-				if err == nil && rc != nil {
-					if rc.Spec.Replicas != nil && *rc.Spec.Replicas < minReplica {
-						return []*apiv1.Pod{}, []*apiv1.Pod{}, &BlockingPod{Pod: pod, Reason: MinReplicasReached}, fmt.Errorf("replication controller for %s/%s has too few replicas spec: %d min: %d",
-							pod.Namespace, pod.Name, rc.Spec.Replicas, minReplica)
-					}
-					replicated = true
-				} else {
-					return []*apiv1.Pod{}, []*apiv1.Pod{}, &BlockingPod{Pod: pod, Reason: ControllerNotFound}, fmt.Errorf("replication controller for %s/%s is not available, err: %v", pod.Namespace, pod.Name, err)
-				}
-			} else {
+		// vadasambar: The else part of this code hasn't been abstracted into
```

Comment:
I disagree with this comment. The fact that something is going to be eventually deleted isn't a good argument for making an extra long function even longer. Firstly, "near future" here means at least months, so it will be around for quite some time. Secondly, eventual removal might actually be cleaner if we can delete the whole function at once.

Reply:
I was hesitant about putting extra time and effort to abstract it out since we are going to delete it anyway but I see your point. Will abstract it out.

Reply:
Abstracted the code into a function. Thank you for the comment.